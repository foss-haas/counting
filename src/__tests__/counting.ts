import { counting } from "..";

test("returns an index and value for each value of the generator", () => {
  const iterable = ["a", "b", "c", "d"];
  let i = 0;
  for (const value of counting(iterable)) {
    expect(value[0]).toEqual(i);
    expect(value[1]).toEqual(iterable[i]);
    i++;
  }
  expect(i).toEqual(iterable.length);
});

test("works with arbitrary generators", () => {
  function* generator() {
    yield "a";
    yield { hello: "world" };
    yield ["hey", "yes", "you"];
    yield true;
    yield null;
    yield undefined;
    yield 123;
  }
  expect(Array.from(counting(generator()))).toMatchSnapshot();
});
