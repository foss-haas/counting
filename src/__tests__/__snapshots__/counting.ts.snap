// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with arbitrary generators 1`] = `
Array [
  Array [
    0,
    "a",
  ],
  Array [
    1,
    Object {
      "hello": "world",
    },
  ],
  Array [
    2,
    Array [
      "hey",
      "yes",
      "you",
    ],
  ],
  Array [
    3,
    true,
  ],
  Array [
    4,
    null,
  ],
  Array [
    5,
    undefined,
  ],
  Array [
    6,
    123,
  ],
]
`;
